sdk/core/Azure.Core/src/ResponseClassifier.cs

```diff
@@ -16,7 +16,14 @@ public class ResponseClassifier
         /// </summary>
         public virtual bool IsRetriableResponse(HttpMessage message)
         {
-            return message.Response.Status == 429 || message.Response.Status == 503;
+            var responseStatus = message.Response.Status;
+
+            return responseStatus == 408 ||
+                   responseStatus == 429 ||
+                   responseStatus == 500 ||
+                   responseStatus == 502 ||
+                   responseStatus == 503 ||
+                   responseStatus == 504;
```

Comment:
Small request for anyone who ends up here when they're having a very bad day...  could we add a comment with the friendly name of each of these status codes right next to it?